test/es6/generator-jit-bugs.js

```diff
@@ -90,4 +90,51 @@ check(gen4.next().value, 1);
 check(gen4.next().value, 2);
 check(gen4.next().value, 3);
 
+title("Load Scope Slots in presence of for-in");
```

Comment:
Note all these cases could be done as async functions (swap yield to await) - testing only as generators as internally the important code pathways are identical and doing it with generators means the test case doesn't have to wait for promises to resolve.